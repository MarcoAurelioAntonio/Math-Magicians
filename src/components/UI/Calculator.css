.number {
  padding: 15px 40px;
  font-size: 25px;
  font-weight: 600;
  background-color: #b3b3b3;
  border: 1px solid #737373;
  position: absolute; /* delete after */
}
