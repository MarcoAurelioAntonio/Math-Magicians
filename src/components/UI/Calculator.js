import React from 'react';
import PropTypes from 'prop-types';

function Button({ buttonText, handleClick }) {
  const isOperator = (valor) => Number.isNaN(valor) && (valor !== '.') && (valor !== '=');

  return (
    <div
      className={`button-container ${isOperator(buttonText) ? 'operator' : ''}`.trimEnd()}
      onClick={handleClick}
      aria-hidden="true"
    >
      {buttonText}
    </div>
  );
}

Button.propTypes = {
  buttonText: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Button;
