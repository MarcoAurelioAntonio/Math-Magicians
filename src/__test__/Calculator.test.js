import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';

describe('Check if Calculator is rendering', () => {
  test('Renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});