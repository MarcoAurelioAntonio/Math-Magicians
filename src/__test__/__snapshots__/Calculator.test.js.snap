// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Calculator is rendering Renders correctly 1`] = `
<div
  className="app-calc"
>
  <h2
    className="title"
  >
    Let\`s do some Maths!!
  </h2>
  <div
    className="calculator-container"
  >
    <div
      className="screen"
    >
      0
    </div>
    <div
      className="buttons-group"
    >
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        AC
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        +/-
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        %
      </div>
      <div
        aria-hidden="true"
        className="button-container operator"
        onClick={[Function]}
      >
        /
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        7
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        8
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        9
      </div>
      <div
        aria-hidden="true"
        className="button-container operator"
        onClick={[Function]}
      >
        x
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        4
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        5
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        6
      </div>
      <div
        aria-hidden="true"
        className="button-container operator"
        onClick={[Function]}
      >
        +
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        1
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        2
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        3
      </div>
      <div
        aria-hidden="true"
        className="button-container operator"
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      className="bottom-group"
    >
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        0
      </div>
      <div
        aria-hidden="true"
        className="button-container"
        onClick={[Function]}
      >
        .
      </div>
      <div
        aria-hidden="true"
        className="button-container operator"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;
